### PR TITLE
Networking/CTP-69-Transfering-Networking-Scripts-To-Solution

### PR DESCRIPTION
**Pull request by erendgrmnc**
- Added ability to start to host a game as server or join a server as a client using "TutorialGame" scene.
- Demo: 

https://github.com/Nijelous/CSC8508-2023/assets/57503158/69cf2956-532c-4f8e-a9b1-9c21b5a32b03

### CODE REVIEW

**Review by Nijelous — CHANGES_REQUESTED** [review]
Code looks solid but I see a lot of variables that don't match the coding standards. I know it's a pain to fix them all but it's worth it to keep the codebase consistent.

File: CSC8503/Main.cpp [hunk 1]
```diff
@@ -40,7 +41,27 @@ int main() {
 	w->ShowOSPointer(false);
 	w->LockMouseToWindow(true);
 
-	TutorialGame* g = new TutorialGame();
+	TutorialGame* g = nullptr;
+	//erendgrmnc: make the bool below true for network test.
+	bool isNetworkTestActive = false;
+	bool isServer = false;
+	if (isNetworkTestActive)
+	{
```
Comment (benblack90) [inline CSC8503/Main.cpp:49]: curly braces should be on the same line

File: CSC8503CoreClasses/GameServer.h [hunk 2]
```diff
@@ -13,17 +13,20 @@ namespace NCL {
...
 			bool SendVariableUpdatePacket(VariablePacket& packet);
+			bool GetPeer(int peerNumber, int& peerId) const;
 
 			virtual void UpdateServer();
 
 		protected:
 			int			port;
 			int			clientMax;
 			int			clientCount;
+			int*        peers;
```
Comment (benblack90) [inline CSC8503CoreClasses/GameServer.h:29]: Member variables should begin with a lower-case m; so mPeers

**Review by benblack90 — CHANGES_REQUESTED** [review]
Just some style stuff, really!

**Review by benblack90 — APPROVED** [review]

File: CSC8503CoreClasses/GameServer.h (hunk 2 above)
Comment (erendgrmnc) [reply CSC8503CoreClasses/GameServer.h]: Done 👍

File: CSC8503/Main.cpp (hunk 1 above)
Comment (erendgrmnc) [reply CSC8503/Main.cpp]: Done 👍

**Review by Nijelous — APPROVED** [review]

**Pull request merged**